CMakeLists.txt

```diff
@@ -1,47 +1,42 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.1)
+
 project(EcoTracker)
-set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_LIST_DIR})
-include_directories(${CMAKE_CURRENT_LIST_DIR})
 
 SET(CMAKE_CXX_STANDARD 11)
-SET(CMAKE_CXX_FLAGS "-O2 -lm")
+SET(CMAKE_CXX_FLAGS "-O2")
 
 find_package(OpenCV REQUIRED)
 include_directories(${OpenCV_DIRS})
 
-file(GLOB eco_srcs
-	${CMAKE_CURRENT_LIST_DIR}/*.hpp
-	${CMAKE_CURRENT_LIST_DIR}/*.h
-	${CMAKE_CURRENT_LIST_DIR}/eco.cc
-	${CMAKE_CURRENT_LIST_DIR}/feature_extractor.cc
-	${CMAKE_CURRENT_LIST_DIR}/feature_operator.cc
-	${CMAKE_CURRENT_LIST_DIR}/ffttools.cc
-	${CMAKE_CURRENT_LIST_DIR}/fhog.cc
-	${CMAKE_CURRENT_LIST_DIR}/interpolator.cc
-	${CMAKE_CURRENT_LIST_DIR}/metrics.cc
-	${CMAKE_CURRENT_LIST_DIR}/optimize_scores.cc
-	${CMAKE_CURRENT_LIST_DIR}/regularization_filter.cc
-	${CMAKE_CURRENT_LIST_DIR}/sample_update.cc
-	${CMAKE_CURRENT_LIST_DIR}/scale_filter.cc
-	${CMAKE_CURRENT_LIST_DIR}/training.cc
-	${CMAKE_CURRENT_LIST_DIR}/EcoTrackerExtern.cc)
+find_package(Threads REQUIRED)
+
+set(ECO_SRCS 
+	eco.cc						eco.hpp
```

Comment:
According to some of my previous experiences, by including headers Visual Studio's class diagram tool can generate class hierarchies correctly while otherwise it won't.

Since including these headers doesn't break anything else, I include them in case someone wants to generate a diagram or etc.